### ISSUE TITLE
Change Detection using MoCo SSL for S1 SAR 2 bands

### ISSUE THREAD

**Issue by vaasuSS** [zhu-xlab/SSL4EO-S12#7]
Hi. Thanks for your awesome work and making this open source. 

I am particularly interested in using the pre-trained model `MoCo-ResNet50-S1-SAR-2-bands` and integrating it with [transfer_change_detection](https://github.com/zhu-xlab/SSL4EO-S12/tree/main/src/benchmark/transfer_change_detection) using the `OSCD` dataset. 

1. What exactly are the preprocessing steps that are done when training the MoCo SSL backbone just using the S1 bands?
2. When integrating this pre-trained model with transfer_change_detection, are there any necessary steps that should be taken before training using OSCD dataset?

Thanks very much for your help.

**Comment by wangyi111** [comment]
Thanks for your question.

> What exactly are the preprocessing steps that are done when training the MoCo SSL backbone just using the S1 bands?

We suppressed extreme values, and normalized the dB scale SAR images with mean and standard deviation for each channel. 

https://github.com/zhu-xlab/SSL4EO-S12/blob/8b943466396c1eedb031f7175db0a0df167d0bc0/src/benchmark/pretrain_ssl/datasets/SSL4EO/ssl4eo_dataset.py#L104

> When integrating this pre-trained model with transfer_change_detection, are there any necessary steps that should be taken before training using OSCD dataset?

We only used optical images for our OSCD experiments, but for SAR should be similar. It could be helpful to supress extreme values and normalize the pixel values either with OSCD mean and std, or mean and std from our dataset.

Hope the above would be helpful.

**Comment by vaasuSS** [comment]
Thanks very much for your feedback. I used the mean and std values from OSCD dataset and updated the code to read and train using OSCD S1 Bands. But there is something wrong during the training. After training for five epochs, the val/precision and val/recall becomes completely zero as shown below. Your feedback is greatly appreciated. Thanks.

`Epoch 4: 100%|██████| 27/27 [05:41<00:00, 12.65s/it, loss=0.182, v_num=0, train/loss=0.172, val/loss=0.227, val/precision=0.000, val/recall=0.000, val/f1=0.000, train/precision=0.00283, train/recall=0.000221, train/f1=0.000411]
`

Below are the screenshots from tensorboard logger

![image](https://user-images.githubusercontent.com/60352460/233206019-e33d830f-c235-4781-85b0-e5e5dce187ec.png)

![image](https://user-images.githubusercontent.com/60352460/233206079-660289b3-7d1d-4a14-8083-aa11481f8e5c.png)

**Comment by wangyi111** [comment]
Does it work with training from scratch without loading a pre-trained model? I didn't test it on SAR-only, guess you have to tune the hyperparameters a bit.

**Comment by vaasuSS** [comment]
Thanks very much @wangyi111. Increasing the epochs did the job but it seems something is still wrong in the training.
1. Can you suggest some of the hyperparameters that needs to be changed?
2. Is there any way I can see the network architecture (similar to keras `model.summary()` or like a `keras.utils.plot_model()`) for the below two models?

https://github.com/zhu-xlab/SSL4EO-S12/blob/8b943466396c1eedb031f7175db0a0df167d0bc0/src/benchmark/transfer_change_detection/main_oscd.py#L49

https://github.com/zhu-xlab/SSL4EO-S12/blob/8b943466396c1eedb031f7175db0a0df167d0bc0/src/benchmark/transfer_change_detection/models/segmentation.py#L17-L18
